boot.php

```diff
@@ -785,60 +785,85 @@ function get_scheme() {
 		return($this->scheme);
 	}
 
+	/**
+	 * @brief Retrieves the Friendica instance base URL
+	 *
+	 * Caches both SSL and non-SSL version for performance
+	 *
+	 * @param bool $ssl
+	 * @return string
+	 */
 	function get_baseurl($ssl = false) {
 
 		// Is the function called statically?
-		if (!is_object($this))
-			return(self::$a->get_baseurl($ssl));
+		if (!is_object($this)) {
+			return self::$a->get_baseurl($ssl);
+		}
 
-		$scheme = $this->scheme;
+		if (!isset($this->baseurl[$ssl ? 'https' : 'http'])) {
```

Comment:
> The parameter doesn't really distinguish between "http" and "https"

I don't get that. Which parameter? If it's the cache key, it cannot distinguish more between "http" and "https" since it will take either of those exact values. If you're talking about the `$ssl` parameter that is misleading since `$ssl = false` can produce a `https://` link then having `$this->baseurl['true']` and `$this->baseurl['false']` is even more misleading.

In any case I don't see a good reason to change the cache index value.